prosopopee/prosopopee.py

```diff
@@ -229,10 +229,155 @@ def main():
 
     if settings["settings"].get("gm"):
         SETTINGS["gm"].update(settings["settings"]["gm"])
+    return settings
+
+
+def build_gallery(gallery, settings, templates, parent_galleries=False):
+
+    if parent_galleries:
+        gallery_path = os.path.join(parent_galleries, gallery)
+    else:
+        gallery_path = gallery
+
+    gallery_settings = yaml.safe_load(open(os.path.join(os.getcwd(), gallery_path, "settings.yaml"), "r"))
+
+    error(isinstance(gallery_settings, dict), "Your %s should be a dict" % (os.path.join(gallery, "settings.yaml")))
+    error(gallery_settings.get("title"), "You should specify a title in %s" % (os.path.join(gallery, "settings.yaml")))
+
+    gallery_index_template = templates.get_template("gallery-index.html")
+    page_template = templates.get_template("page.html")
+    gallery_cover = {}
+
+    dirs = filter(lambda x: x not in (".", "..") and os.path.isdir(os.path.join(gallery_path, x)) and
+                  os.path.exists(os.path.join(os.getcwd(), gallery_path, x, "settings.yaml")),
+                  os.listdir(os.path.join(os.getcwd(), gallery_path)))
+
+    if not os.path.exists(os.path.join("build", gallery_path)):
+            os.makedirs(os.path.join("build", gallery_path))
+
+    if gallery_settings.get("public", True) or dirs:
+        error(gallery_settings.get("title"), "Your gallery describe in %s need to have a "
+                                             "title" % (os.path.join(gallery, "settings.yaml")))
+        error(gallery_settings.get("cover"), "You should specify a path to a cover picture "
+                                             "in %s" % (os.path.join(gallery, "settings.yaml")))
+
+        if isinstance(gallery_settings["cover"], dict):
+            cover_image_path = os.path.join(gallery_path, gallery_settings["cover"]["name"])
+            cover_image_url = os.path.join(gallery, gallery_settings["cover"]["name"])
+            cover_image_type = gallery_settings["cover"]["type"]
+        else:
+            cover_image_path = os.path.join(gallery_path, gallery_settings["cover"])
+            cover_image_url = os.path.join(gallery, gallery_settings["cover"])
+            cover_image_type = "image"
+
+        error(os.path.exists(cover_image_path), "File for %s cover image doesn't exist at "
+                                                "%s" % (gallery, cover_image_path))
+
+        gallery_cover = {
+            "title": gallery_settings["title"],
+            "link": gallery,
+            "sub_title": gallery_settings.get("sub_title", ""),
+            "date": gallery_settings.get("date", ""),
+            "tags": gallery_settings.get("tags", ""),
+            "cover_type": cover_image_type,
+            "cover": cover_image_url,
+        }
+
+        if dirs:
+            subgallery_theme = gallery_settings.get("theme")
```

Comment:
I think that everything related to theme can be extracted in a `get_theme` function or something like that.